packages/babel-plugin-transform-typescript/src/index.js

```diff
@@ -44,11 +44,90 @@ function registerGlobalType(programScope, name) {
 }
 
 export default declare(
-  (api, { jsxPragma = "React", allowNamespaces = false }) => {
+  (
+    api,
+    { jsxPragma = "React", allowNamespaces = false, declareFields = false },
+  ) => {
     api.assertVersion(7);
 
     const JSX_ANNOTATION_REGEX = /\*?\s*@jsx\s+([^\s]+)/;
 
+    const classMemberVisitors = {
+      field(path) {
```

Comment:
I didn't use the `ClassProperty` visitor because the class fields transform transforms them using the `Class` visitor, so they would never be visited in practice.